lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -1058,9 +1058,13 @@ public function createSchemaConfig()
         }
 
         $params = $this->_conn->getParams();
-        if (isset($params['defaultTableOptions'])) {
-            $schemaConfig->setDefaultTableOptions($params['defaultTableOptions']);
+        if (! isset($params['defaultTableOptions'])) {
+            $params['defaultTableOptions'] = [];
         }
+        if (! isset($params['defaultTableOptions']['charset']) && isset($params['charset'])) {
```

Comment:
Should we test for emptiness ?

Reply:
I guess if you set $params['charset'] to null, '', [], an object or ... you are deliberately trying to break stuff. Do I understand your question right?